src/main/java/io/appium/java_client/proxy/ProxyListenersContainer.java

```diff
@@ -76,27 +66,7 @@ public <T> T setListeners(T proxyInstance, Collection<MethodCallListener> listen
             throw new RuntimeException(e);
         }
         try {
-            int i = 0;
-            boolean wasInstancePresent = false;
-            while (i < this.listeners.size()) {
-                Pair<WeakReference<?>, Collection<MethodCallListener>> pair = this.listeners.get(i);
-                Object key = pair.getKey().get();
-                if (key == null) {
-                    // The instance has been garbage-collected
-                    this.listeners.remove(i);
-                    continue;
-                }
-
-                if (key == proxyInstance) {
-                    pair.getValue().clear();
-                    pair.getValue().addAll(listeners);
-                    wasInstancePresent = true;
-                }
-                i++;
-            }
-            if (!wasInstancePresent) {
-                this.listeners.add(new Pair<>(new WeakReference<>(proxyInstance), new HashSet<>(listeners)));
-            }
+            this.listeners.put(String.valueOf(proxyInstance.getClass()), new HashSet<>(listeners));
```

Comment:
I am not convinced that class itself could be the key here. listeners must be connected to instances

Reply:
Also, why it is necessary to perform `String.valueOf` on the value? What would be the example output?

Reply:
> 

Removed String.valueOf